### ISSUE TITLE
Issue startupinfo_winerror87 attribute error

### ISSUE THREAD

**Issue by MaximilianSP** [majroy/pyCM#8]
When performing Generate Mesh, following error occurs: 

  File "C:\Users\xxxxxxx\anaconda3\lib\site-packages\pyCM\preprocess.py", line 306, in <lambda>
    self.ui.meshscriptButton.clicked.connect(lambda: self.WriteGeo())

  File "C:\Users\xxxxxxx\anaconda3\lib\site-packages\pyCM\preprocess.py", line 1136, in WriteGeo
    self.RunMeshScript()

  File "C:\Users\xxxxxxx\anaconda3\lib\site-packages\pyCM\preprocess.py", line 868, in RunMeshScript
    out=sp.check_output([execStr,"-3",self.geofile,"-o",self.vtkFile])

  File "C:\Users\xxxxxxx\anaconda3\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout

  File "C:\Users\xxxxxxx\anaconda3\lib\subprocess.py", line 403, in run
    with Popen(*popenargs, **kwargs) as process:

  File "C:\Users\xxxxxxx\anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 171, in __init__
    super(SubprocessPopen, self).__init__(*args, **kwargs)

  File "C:\Users\xxxxxxx\anaconda3\lib\subprocess.py", line 709, in __init__
    restore_signals, start_new_session)

  File "C:\Users\xxxxxxx\anaconda3\lib\subprocess.py", line 997, in _execute_child
    startupinfo)

OSError: [WinError 87] The Parameter is incorrect

Python: 3.7

Do you know what the reason for this error might be? Thank you in advance for your help.

**Comment by majroy** [comment]
It appears that you are running this through Spyder, it's not something that I've tried. I also haven't tested things with the Anaconda ecosystem.
1) Which version of pyCM, gmsh and windows are you running?
2) Does the same error occur with Abaqus?
3) Does it work outside of Spyder?

I've just checked and have found that meshing and job submission with the version of 'preprocess' in v1.1.0 on Win10, Python 3.7, Gmsh 4 and Abaqus 2018.

**Comment by MaximilianSP** [comment]
I am running the Code with:
- pyCM-1.1.0, python 3.6.7, gmsh 4.0.7, Windows 10
- Unfortunately I don't have Abaqus for now
- When I run it via the anaconda prompt, python crashes when trying to execute the meshing
I will upgrade python to the newest Version to see whether it will work.

**Comment by MaximilianSP** [comment]
The line in preprocess.py 868 (RunMeshScript):
out=sp.check_output([execStr,"-3",self.geofile,"-o",self.vtkFile]). I was told to check whether all the arguments are strings and that the bug might occur because of non valid string arguments.

**Comment by majroy** [comment]
Please check if the latest version 1.1.1 fixes the issue - there is now a call to run check_output in a shell.

If it doesn't, then you can manually run gmsh. When you press the execute button on the GUI, two main things occur: first, a *.geo file is generated which contains instructions for gmsh to generate a VTK file, and then a call to the command line and execute gmsh.exe with the *.geo file.  Run the analysis to the point where you press the execute button, kill pyCM, run gmsh, and then re-load pyCM with the applicable *.mat file when a valid VTK file is generated. You will have to adopt a similar approach when running Calculix.

**Comment by MaximilianSP** [comment]
Thank you for the Input! With the new issue, I get a different error:
 
  File "C:\Users\msprenge\anaconda3\lib\site-packages\pyCM\preprocess.py", line 306, in <lambda>
    self.ui.meshscriptButton.clicked.connect(lambda: self.WriteGeo())

  File "C:\Users\msprenge\anaconda3\lib\site-packages\pyCM\preprocess.py", line 1139, in WriteGeo
    with open(self.vtkFile, 'r') as file: vtkcontents=file.read()

FileNotFoundError: [Errno 2] No such file or directory: 'C:/Users/msprenge/.anaconda/pyCM-master/pyCM-master/exampledata/Test10012019/genmesh.vtk'

The message in the GUI says: " No CAE Input file specified...Idle". According to the preprocess code, the message that is displayed results from using the Abaqus Routine. Is there an automatic switch from the gmsh to abaqus routine implemented? 

I will try to run the process manually. Thank you again for your help.

**Comment by majroy** [comment]
The error states that your vtk file 'genmesh.vtk' can't be read. The error has nothing to do with Abaqus.

**Comment by MaximilianSP** [comment]
I agree with you. So I don't understand why the message " No CAE Input file specified...Idle" is shown. I have attached a screenshot. <img width="937" alt="no cae file" src="https://user-images.githubusercontent.com/45238826/50972533-3161ba80-14e7-11e9-8a43-78b500136693.PNG">

Unfortunately, no vtk file is generated at this stage. This is the end of the geo file that is generated.

Line(278) = {278,1};
Line Loop(279) = {1:278};
Plane Surface(280) = {279};
Recombine Surface {280};

OutOfPlane[]= Extrude {0, 0, 161.00000000} {
 Surface{280};
 Layers{ {1,1,1,1,1,1,1,1,1,1}, {0.0122,0.0297,0.0548,0.0907,0.1421,0.2159,0.3217,0.4732,0.6904,1.0000} };
 Recombine;};
 
//EOF

**Comment by majroy** [comment]
The message isn't wrong - no CAE file was specified. If you are getting a *.geo file, but no mesh (in the form of a VTK file) then gmsh is either not being called, or not executing (correctly). Returns from the shell which would indicate why are probably being masked; potentially because you're running Anaconda. Create the VTK file yourself by running gmsh with the *.geo file; see above.

**Comment by MaximilianSP** [comment]
Now on linux, the issue I get by producing the vtk file in gmsh externally is: 

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/Annotation/vtkAxisActor.cxx, line 1872
vtkAxisActor (0x55e691350880): Number of labels -2147483647 is invalid

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/Annotation/vtkAxisActor.cxx, line 1872
vtkAxisActor (0x55e69152b4d0): Number of labels -2147483647 is invalid

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/Annotation/vtkAxisActor.cxx, line 1872
vtkAxisActor (0x55e691639170): Number of labels -2147483647 is invalid

ERROR: In /work/standalone-x64-build/VTK-source/Rendering/Annotation/vtkAxisActor.cxx, line 1872
vtkAxisActor (0x55e69175ef70): Number of labels -2147483647 is invalid

Have you encountered such an issue before? Maybe I have used a non compatible mesh in gmsh.

**Comment by majroy** [comment]
Email me a *.mat file containing your analysis up to the pre-processing step.

**Comment by MaximilianSP** [comment]
Found the issue! Although I added the folder with the gmsh .exe file to the path, it was necessary to point to the folder as first step in preprocess.py. Then it worked! I just should have read more carefully. Issue is solved : ) Thank you again for the quick responses and help.
